src/models/blockchain/block/mod.rs

```diff
@@ -464,32 +463,44 @@ impl Block {
+        if matches!(network, Network::Alpha) {
```

Comment:
My only intent here was to cleanup the fn so it is clearer what is happening and make it more correct.  Upon review, it seems I introduced a problem along the way though.

Previously the premine_distribution() fn did not accept a network arg.  Thus regardless of the current network it would generate a devnet wallet address that receives 20 coins.   It then adds 2 alphanet addresses and 3 mainnet addresses, all with varying amounts.

The genesis_block() fn calls premine_distribution() and then sums the amounts:

```
        let premine_distribution = Self::premine_distribution();
        let total_premine_amount = premine_distribution
            .iter()
            .map(|(_receiving_address, amount)| *amount)
            .sum();
```

So right away this seems like a (serious?) bug because if we are on mainnet it is including amounts from 2 alphanet addresses, and vice-versa.

aside: Perhaps of even greater concern is that the genesis_block() fn adds utxos containing addresses from the wrong network into the genesis block of each network type without throwing any error or panic.  @aszepieniec I think we perhaps ought to review how this is possible and make it not possible.  (outside scope of this PR though.)

Another consideration is that all addresses inside premine_distribution() presently are just placeholders intended for testing.   The real addresses/amounts are to be added later.

So I addressed the above (excluding genesis-block validation) by:

1. introduce a network arg to the premine_distribution() fn so that it only returns addresses/amounts for the current network.
2. compile the current premine_distribution() fn only for unit tests, and add an empty fn for real builds.  The real addresses can be placed in this fn.

Unfortunately I did not notice that there are actually three mainnet addresses in the test data.  I thought they were all Alphanet, and thus added the three mainnet addrs inside the Alpha block.   I will fix that.

Reply:
> Unfortunately I did not notice that there are actually three mainnet addresses in the test data. I thought they were all Alphanet, and thus added the three mainnet addrs inside the Alpha block. I will fix that.

fixed in 83aa3eca45da1ec659fb7288ef787787fdbb039e